test/screenshot/README.md

```diff
@@ -1,22 +1,144 @@
 # Screenshot Tests
 
-## Local dev server
+Prevent visual regressions by running screenshot tests on every PR.
 
-Run:
+## Quick start
+
+Add the following to your `~/.bash_profile` (Mac) or `~/.bashrc` (Linux):
+
+```bash
+export CBT_USERNAME='you@example.com'
+export CBT_AUTHKEY='example'
+export MDC_GCLOUD_SERVICE_ACCOUNT_KEY_FILE_PATH='path/to/gcp-credentials.json'
+```
+
+In a new terminal, run:
+
+```bash
+npm run screenshot:test
+```
+
+This will take several minutes to complete.
+
+When the tests finish, you should see a link to the diff report page in the terminal:
+
+```
+Found 0 screenshot diffs!
+https://storage.googleapis.com/mdc-web-screenshot-tests/advorak/2018/05/22/17_49_05_673/d3a549bb4/report.html
+```
+
+Next, modify a Sass file, and rerun the tests:
+
+```bash
+echo '.mdc-button:not(:disabled){color:red}' >> packages/mdc-button/mdc-button.scss
+npm run screenshot:test
+```
+
+You should see something like this in the terminal:
+
+```
+Found 65 screenshot diffs!
+https://storage.googleapis.com/mdc-web-screenshot-tests/advorak/2018/05/22/16_23_58_618/af3d7271f/report.html
+```
+
+## Basic usage
+
+### Updating "golden" screenshots
+
+```bash
+npm run screenshot:update-goldens
+```
+
+In addition to running the tests, this command will also update your local `test/screenshot/golden.json` file with the 
+newly captured screenshots.
+
+### Rerunning flaky tests
```

Comment:
Given we're now auto-retrying, I'd re-word this to "Rerunning a subset of tests" and re-word below to "You can rerun a subset of tests without running the entire suite, filtering by browser and/or URL:"

I'm also wondering if it's better to show this example with `screenshot:test`, though it does work with both... ostensibly we should be running the entire suite on a branch at least once though, right?

Reply:
Tangentially, I've been wondering if we have suggestions on recommended workflow that can be documented. Like, screenshot:test and screenshot:update-goldens are both going to run the tests... and presumably we should run the tests once to confirm the results before updating goldens... but then we have to wait for all of the tests to run AGAIN when we run update-goldens. I guess if you haven't changed anything you can then filter update-goldens based on what was actually found in test, but that's subject to user error.

As much as I don't want to add more work, it might be nice to eventually repurpose update-goldens to hand it an existing report and update the JSON based on that rather than needing to run all of the tests all over again?

Reply:
Yeah, all CLI flags work with both `screenshot:test` and `screenshot:update-goldens`.

As far as workflow goes, I was planning to eventually add an "Approve" button on the diff report page. It would copy a CLI command to the clipboard that updates `golden.json` without rerunning the tests.

To do that, I would change `update-goldens` so that it doesn't run any tests, but instead accepts a URL or file path of a `snapshot.json` file, which would be automatically generated by `screenshot:test` and uploaded to GCS.

So the workflow would go like this:

```bash
# ... hack hack hack ...

npm run screenshot:test

# ... user views the report and clicks the "Approve" button ...

npm run screenshot:update -- https://storage.../snapshot.json

git add test/screenshot/golden.json
git commit -m 'WIP: Update goldens'
```

This should be pretty quick and easy to do.